spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -88,6 +88,11 @@ sealed class SpdxExpression {
         }
     }
 
+    /**
+     * Deep-copy the whole license expression object.
+     */
+    abstract fun copy(): SpdxExpression
```

Comment:
To not lose the default copy function of the data classes, what about this instead?

```kotlin
    fun clone() = when (this) {
        is SpdxCompoundExpression -> copy()
        is SpdxLicenseExceptionExpression -> copy()
        is SpdxLicenseIdExpression -> copy()
        is SpdxLicenseReferenceExpression -> copy()
    }
```

Reply:
Or you could do `fun clone() = SpdxExpression.parse(toString())`.

Reply:
> Or you could do `fun clone() = SpdxExpression.parse(toString())`.

Don't you have to remember the `strictness` for that?

Reply:
Very good idea to use a `when` in the base class. However, *if* we do this at all, I'd prefer the function to be also called `copy()`, so I had to name the arguments again.

Reply:
@maxhbr The strictness is not stored in the expression object, so using the default of `ALLOW_ANY` should be fine, but it would probably be better to explicitly set it, as in `SpdxExpression.parse(toString(), Strictness.ALLOW_ANY`.

Reply:
@sschuberth  I thought we would lose the default copy function, but was wrong as we do not use the signature of the generated functions, so I'm fine with using the name `copy`.